docs/reference/metrics.md

```diff
@@ -51,11 +51,12 @@ Note that `Core` includes the Typescript SDK.
 | request_latency                           | gRPC request latency                                                                                                                                            | Histogram   | Core-Client, Go-Client, Java-Service-Client | namespace, operation                                |
 | sticky_cache_hit                          | A workflow task found a cached workflow to run against                                                                                                          | Counter     | Core, Go, Java                              | namespace, task_queue                               |
 | sticky_cache_miss                         | A workflow task did not find a cached workflow                                                                                                                  | Counter     | Core, Go, Java                              | namespace, task_queue                               |
-| sticky_cache_size                         | Current cache size in number of workflows                                                                                                                       | Gauge       | Core, Go, Java                              | namespace, task_queue                               |
+| sticky_cache_size                         | Current cache size in number of workflows                                                                                                                       | Gauge       | Core, Go, Java                              |                                                     |
```

Comment:
@Sushisource Could you please help me to verify that the metrics changes are aligned with Core?

This one is especially important, as Workflow Cache Size should be per Workers host, not per namespace and task queue. This metric/limit is very different and exists to limit the total resource consumption between all workers.

Reply:
This is per namespace/tq. Specifically it is per worker.

Core resolves a problem that exists in Go/Java relating to controlling max cached workflows. Go and Java use one sticky queue for the host. This makes it impossible to accurately constrain the number of workflow slots to be <= the defined maximums (I can dig up where I wrote the explanation why, but, this is true).

So, Core fixes this by using unique sticky queues per normal queue.

Now, this doesn't mean that we can't *also* make a host-wide metric for this, so Core could add that, but cache limits in the config need to stay per-worker in Core.

Reply:
As we discussed, there is no problem in Go or Java with controlling max cached workflows. There is a tricky situation with a shared poller (in JavaSDK) and an amount of executing threads, but it's not really related to cached workflow limit.
Because adjustment of maximum cached workflows should be considered when we are out (or have a surplus) of host level resources, like RAM and total system threads count, this metric mostly makes sense per process and the limit also should be enforced per process to avoid underutilization.
But for now, I just document the state as it is.